gpiozero/devices.py

```diff
@@ -265,13 +260,13 @@ def _release_all(self):
         :meth:`_release_pins` for further information).
         """
         with self._res_lock:
-            Device._reservations = {
+            Device._reservations = defaultdict(list, {
                 address: [
                     ref for ref in conflictors
                     if ref() not in (self, None)
                     ]
                 for address, conflictors in self._reservations.items()
-                }
+                })
```

Comment:
Precisely - just re-writes the dict with all references to self (and dead-refs) removed; easiest way to do it (and safe given it's all done under `_res_lock`).